src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckPolicyMessageContentCollector.java

```diff
@@ -72,47 +137,48 @@ protected void addCategoryItems(final List<CategoryItem> categoryItems, final Js
         }
 
-        final ItemOperation operation;
-        try {
-            operation = getOperationFromNotification(notificationContent);
-        } catch (final IllegalArgumentException e) {
-            logger.error("Unrecognized notification type", e);
+        final ItemOperation operation = getOperationFromNotification(notificationContent);
+        if (operation == null) {
             return;
         }
+
         for (final LinkableItem policyItem : policyItems) {
             final String policyUrl = policyItem.getUrl().orElse("");
-            addApplicableItems(categoryItems, notificationContent.getId(), policyItem, policyUrl, operation, combinedNameItem, applicableItems);
+            addApplicableItems(categoryItems, notificationContent.getId(), policyItem, policyUrl, operation, applicableItems);
         }
     }
 
-    protected ItemOperation getOperationFromNotification(final NotificationContent notificationContent) {
+    private ItemOperation getOperationFromNotification(final NotificationContent notificationContent) {
+        final ItemOperation operation;
         final String notificationType = notificationContent.getNotificationType();
         if (NotificationType.RULE_VIOLATION_CLEARED.name().equals(notificationType)) {
-            return ItemOperation.DELETE;
+            operation = ItemOperation.DELETE;
         } else if (NotificationType.RULE_VIOLATION.name().equals(notificationType)) {
-            return ItemOperation.ADD;
+            operation = ItemOperation.ADD;
         } else if (NotificationType.POLICY_OVERRIDE.name().equals(notificationType)) {
-            return ItemOperation.DELETE;
+            operation = ItemOperation.DELETE;
+        } else {
+            operation = null;
         }
-        throw new IllegalArgumentException(String.format("The notification type '%s' is not valid for this collector.", notificationType));
+        logger.error("Unrecognized notification type: The notification type '{}' is not valid for this collector.", notificationType);
+        return operation;
     }
 
-    private void addApplicableItems(final List<CategoryItem> categoryItems, final Long notificationId, final LinkableItem policyItem, final String policyUrl, final ItemOperation operation,
-        final Optional<LinkableItem> nameItem, final SortedSet<LinkableItem> applicableItems) {
+    private void addApplicableItems(final List<CategoryItem> categoryItems, final Long notificationId, final LinkableItem policyItem, final String policyUrl, final ItemOperation operation, final SortedSet<LinkableItem> applicableItems) {
+        final List<String> keyItems = applicableItems.stream().map(LinkableItem::getValue).collect(Collectors.toList());
+        String[] keyArray = new String[keyItems.size() + 1];
+        keyArray = keyItems.toArray(keyArray);
+        keyArray[keyArray.length - 1] = policyUrl;
+        final CategoryKey categoryKey = CategoryKey.from(CATEGORY_TYPE, keyArray);
```

Comment:
Would it be better to just add a new static initializer that takes a List<String>?